sktime/forecasting/prophet.py

```diff
@@ -0,0 +1,341 @@
+            changepoint_range=changepoint_range,
+            yearly_seasonality=yearly_seasonality,
+            weekly_seasonality=weekly_seasonality,
+            daily_seasonality=daily_seasonality,
+            holidays=holidays,
+            seasonality_mode=seasonality_mode,
+            seasonality_prior_scale=seasonality_prior_scale,
+            holidays_prior_scale=holidays_prior_scale,
+            changepoint_prior_scale=changepoint_prior_scale,
+            mcmc_samples=mcmc_samples,
+            interval_width=interval_width,
+            uncertainty_samples=uncertainty_samples,
+            stan_backend=stan_backend,
+            **kwargs
+        )
+
+    def fit(self, y, X=None, fh=None, **fit_params):
+        """Fit to training data.
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series to which to fit the forecaster.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        y, X = check_y_X(y, X, index_type=pd.DatetimeIndex)
+        self._set_y_X(y, X)
+        self._set_fh(fh)
+
+        # We have to bring the data into the required format for fbprophet:
+        df = pd.DataFrame(y.rename("y"))
+        df["ds"] = y.index
+
+        # Add seasonality
+        if self.add_seasonality:
+            self._forecaster.add_seasonality(**self.add_seasonality)
+
+        # Add country holidays
+        if self.add_country_holidays:
+            self._forecaster.add_country_holidays(**self.add_country_holidays)
+
+        # Add regressor (multivariate)
+        if X is not None:
+            df = df.merge(X, left_index=True, right_on=X.index)
+            for col in X.columns:
+                self._forecaster.add_regressor(col)
+
+        self._forecaster.fit(df=df, **fit_params)
+        self._is_fitted = True
+        return self
+
+    def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        """Predict
+
+        Parameters
+        ----------
+        fh : array-like
+            The forecasters horizon with the steps ahead to to predict.
+            Default is
+            one-step ahead forecast, i.e. np.array([1]).
+        X : pd.DataFrame, optional
+            Exogenous data, by default None
+        return_pred_int : bool, optional
+            Returns a pd.DataFrame with confidence intervalls, by default False
+        alpha : float, optional
+            Alpha level for confidence intervalls, by default DEFAULT_ALPHA
+
+        Returns
+        -------
+        y_pred : pandas.Series
+            Returns series of predicted values.
+
+        Raises
+        ------
+        Exception
+            Error when merging data
+        """
+        if type(fh._values) == pd.DatetimeIndex:
```

Comment:
```suggestion
        if isinstance(fh.to_pandas(), pd.DatetimeIndex):
```

Reply:
Since we're enforcing a `pd.DateTimeIndex` in `fit` for the training data, I'd also enforce it here for the absolute representation of the `fh`, but the `fh` may also be relative, so we need to check `fh.to_absolute()`

Reply:
I actually did not 100% get what `fh.to_absolute()` is doing and why it is relevant to check it here. Would be great if you can explain?

Reply:
The forecasting horizon can be represented in two ways: relative or absolute, that is, (integer-valued) steps ahead *relative* to some reference point (e.g. the end of the training set) or the exact, *absolute* time points to predict (e.g. time stamps). Different libraries expect different specifications. The idea of our `ForecastingHorizon` class is to make it easy to switch between these representations. `fbprophet` expect the absolute one as far as I understand, as they encode it in the (time) index of the future dataframe, and the index is always absolute. To get the absolute representation, you can call inside the forecaster: `fh.to_absolute(self.cutoff)` where the `cutoff` is the reference point that allows you to translate one representation to the other.